client/engine/store/persiststore.go

```diff
@@ -0,0 +1,474 @@
+package store
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/channel"
+	"github.com/statechannels/go-nitro/channel/consensus_channel"
+	"github.com/statechannels/go-nitro/crypto"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/protocols/directdefund"
+	"github.com/statechannels/go-nitro/protocols/directfund"
+	"github.com/statechannels/go-nitro/protocols/virtualdefund"
+	"github.com/statechannels/go-nitro/protocols/virtualfund"
+	"github.com/statechannels/go-nitro/types"
+	"github.com/tidwall/buntdb"
+)
+
+type PersistStore struct {
```

Comment:
Is there a better name for this? `FileStore`? `BuntDbStore`?

Reply:
> * [Sync always: 247 ms TTFP](http://34.168.92.245:3000/d/5OBBeW37k/time-to-first-payment?orgId=1&from=1678478595989&to=1678478699884)
> * [Sync every second: 86.5 ms  TTFP](http://34.168.92.245:3000/d/5OBBeW37k/time-to-first-payment?orgId=1&from=1678478669430&to=1678478825842&var-runId=cg5op0gnr2ghv5ug1m8g&var-jobCount=3&var-testDuration=1m0s&var-hubs=1&var-payees=5&var-jitter=1&var-latency=10&var-payers=2&var-payeepayers=0&var-nitroVersion=v0.0.0-20230310171721-486f70744942&var-storeSyncFrequency=1)
> * [Never Sync Policy: 84.1 ms TTFP](http://34.168.92.245:3000/d/5OBBeW37k/time-to-first-payment?orgId=1&from=1678479249909&to=1678479339937&var-runId=cg5o96gnr2ghv5ug1m60&var-jobCount=3&var-testDuration=1m0s&var-hubs=1&var-payees=5&var-jitter=1&var-latency=10&var-payers=2&var-payeepayers=0&var-nitroVersion=v0.0.0-20230310171721-486f70744942&var-storeSyncFrequency=1)

This is really interesting data to have. Interesting that they only offer these three levels of sync frequency. Going every second seems to have a negligible effect on TTFP (which is great!). I wonder what the TTFP vs sync period graph would look like 🤔 . It might be that sync'ing every 10ms is also having only a small effect. I guess it will be hard for us to know without hacking around with `buntdb`?

Reply:
> Is there a better name for this? `FileStore`? `BuntDbStore`?

I don't have a strong opinion. `BuntDbStore` seems reasonable as this implementation seems fairly tightly coupled to `buntdb`.